torch/nn/utils/parametrize.py

```diff
@@ -0,0 +1,375 @@
+from ..modules.module import Module
+from ..parameter import Parameter
+import functools
+import inspect
+
+
+class Parametrization(Module):
+    r"""A kind of Module that parametrizes a Parameter or a buffer in terms
+    of its forward function
+
+    After registering a Parametrization ``param`` on a tensor ``t`` within a module
+    ``module`` with :meth:`#register_parametrization`, ``module.t`` will be turn
+    into a property, and will return ``param(t)``, where ``t`` is the previous tensor
+
+    After registering a Parametrization, the parametrization will be accessible under
+    ``module.t_param``. The parametrized and unparametrized tensors may be accessed
+    through ``param.evaluate()`` and ``param.original()``
+
+    Before a parametrization is registered, it may be chained to others to compose them
+    via the ``chain`` method. A parametrization may be evaluated on a tensor ``t`` by
+    ``param.evaluate(t)``, even when it has not been registered
+
+    Parametrizations may be registered on parameters, buffers, and already parametrized
+    tensors
```

Comment:
Are tensors in here just any attribute of type `torch.Tensor` that is not necessarily a parameter or a buffer? It reads to me that way, but I don't think that's what you wanted.

Reply:
I will reword what you pointed out here and in the next comment, as we are just dealing with buffers and parameters.

Reply:
I read it again and yes, it may be evaluated in any tensor at any time. For example, if you have a pruning method with certain mask, you may evaluate it on any tensor (provided that the sizes match) via `parametrization.evaluate(t)`.